google_guest_agent/non_windows_accounts_integ_test.go

```diff
@@ -3,10 +3,62 @@
 package main
 
 import (
+	"fmt"
+	"os"
 	"os/exec"
+	"strings"
 	"testing"
 )
 
+const (
+	testUser           = "integration-test-user"
+	defaultgroupstring = "adm,dip,docker,lxd,plugdev,video,google-sudoers"
+)
+
+func TestCreateAndRemoveGoogleUser(t *testing.T) {
+	if exist, err := userExists(testUser); exist == true {
+		t.Fatalf("test user should not exist")
+	}
+	if err := createGoogleUser(testUser); err != nil {
+		t.Errorf("createGoogleUser failed creating test user")
+	}
+	if exist, err := userExists(testUser); exist != true || err != nil {
+		t.Errorf("test user should exist")
+	}
+	cmd := exec.Command("groups", testUser)
+	ret := runCmdOutput(cmd)
+	if ret.ExitCode() != 0 {
+		t.Errorf("failed looking up groups for user: stdout:%s stderr:%s", ret.Stdout(), ret.Stderr())
+	}
+	groups := strings.Split(strings.TrimSpace(strings.Split(ret.Stdout(), ":")[1]), " ")
+	expectedGroups := config.Section("Accounts").Key("groups").MustString(defaultgroupstring)
+	for _, group := range groups {
+		if !contains(group, strings.Split(expectedGroups, ",")) {
+			t.Errorf("test user has been added to an unexpected group %s", group)
+		}
+	}
+	if _, err := os.Stat(fmt.Sprintf("/home/%s", testUser)); err != nil {
+		t.Errorf("test user home directory is not exist")
+	}
+	if err := createGoogleUser(testUser); err == nil {
+		t.Errorf("createGoogleUser did not return error when creating user that already exists")
+	}
+	if err := removeGoogleUser(testUser); err != nil {
+		t.Errorf("removeGoogleUser did not remove user")
+	}
+	if exist, err := userExists(testUser); exist == true {
+		t.Errorf("test user should not exist")
+	}
+	if config.Section("Accounts").Key("deprovision_remove").MustBool(false) {
+		if _, err := os.Stat(fmt.Sprintf("/home/%s", testUser)); err == nil {
+			t.Errorf("test user home directory should not exist")
+		}
+	}
```

Comment:
this won't work unless you initialize the config global

alternatively since we know the execution environment, you can remove this entire section and just test this in the config-manipulating tests we have to add next

```suggestion
```